### DISCUSSION TITLE
Compilation error with OpenFOAM version 2112

### DISCUSSION THREAD

**Discussion by SamSalehian** [WENO-OF/WENOEXT#39] [General]
Hello,

I know the installation description mentioned OpenFOAM (ESI) v1912-v2012 is supported, but I was wondering if you knew of any workaround for compiling against V2112. I tested it and got the following errors:

[  3%] Building CXX object libWENOEXT/CMakeFiles/WENOEXT.dir/BlazeIO/BlazeIO.C.o
cc1plus: error: bad value (‘tigerlake’) for ‘-march=’ switch
cc1plus: note: valid arguments to ‘-march=’ switch are: nocona core2 nehalem corei7 westmere sandybridge corei7-avx ivybridge core-avx-i haswell core-avx2 broadwell skylake skylake-avx512 cannonlake icelake-client icelake-server cascadelake bonnell atom silvermont slm goldmont goldmont-plus tremont knl knm x86-64 eden-x2 nano nano-1000 nano-2000 nano-3000 nano-x2 eden-x4 nano-x4 k8 k8-sse3 opteron opteron-sse3 athlon64 athlon64-sse3 athlon-fx amdfam10 barcelona bdver1 bdver2 bdver3 bdver4 znver1 znver2 btver1 btver2 native
cc1plus: error: bad value (‘tigerlake’) for ‘-mtune=’ switch
cc1plus: note: valid arguments to ‘-mtune=’ switch are: nocona core2 nehalem corei7 westmere sandybridge corei7-avx ivybridge core-avx-i haswell core-avx2 broadwell skylake skylake-avx512 cannonlake icelake-client icelake-server cascadelake bonnell atom silvermont slm goldmont goldmont-plus tremont knl knm intel x86-64 eden-x2 nano nano-1000 nano-2000 nano-3000 nano-x2 eden-x4 nano-x4 k8 k8-sse3 opteron opteron-sse3 athlon64 athlon64-sse3 athlon-fx amdfam10 barcelona bdver1 bdver2 bdver3 bdver4 znver1 znver2 btver1 btver2 generic native
make[2]: *** [libWENOEXT/CMakeFiles/WENOEXT.dir/build.make:76: libWENOEXT/CMakeFiles/WENOEXT.dir/BlazeIO/BlazeIO.C.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:316: libWENOEXT/CMakeFiles/WENOEXT.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

Cheers,

**Comment by JanGaertner** [comment]
Hello Sam,

it seems that your compiler has a problem with the native switch for the CPU architecture. Try switching it off with 
./Allwmake -DMARCH_NATIVE=OFF

**Comment by SamSalehian** [comment]
Hello Jan,

Thanks for your quick response.  I compiled with the -DMARCH_NATIVE=OFF flag, and it compiled perfectly. 

Cheers,
